### PR TITLE
cylc validate: fix failure on XXX-FAM task name

### LINKED ISSUE #1778: cylc validate: family graphing error
Using 6.9.1 (and 6.1.2) the following suite fails validation.

```ini
[scheduling]
    [[dependencies]]
        graph = "baz-foo => bar"
[runtime]
    [[foo]]
    [[bar, baz-foo]]
        inherit = foo
```
cylc validate reports:
```
baz-foo => bar
'ERROR, family triggers must be qualified, e.g. foo:succeed-all'
```
It works fine if
* I use underscore rather than hyphen in the task name
* I  change the family name to anything other than "foo" (even "Foo" works fine")

### PR DESCRIPTION
**Pull request by matthewrmshin**
Close #1778.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark please review.

File: lib/cylc/config.py [hunk 1]
```diff
@@ -1400,7 +1400,12 @@ def process_graph_line(self, line, section, seq, offset_seq_map,
             # backslashed re markers like \b from being interpreted as
             # normal escapeded characters.
 
-            if fam not in line:
+            fam_in_line = re.search(
+                r"(?<!%(name_char_re)s)%(fam)s(?!(name_char_re)s)" % {
```
Comment (hjoliver) [inline lib/cylc/config.py]: Missing '%' in the second `name_char_re`!

Comment (matthewrmshin) [inline lib/cylc/config.py]: Fixed, and added extra test.

**Comment by matthewrmshin** [comment]
Branch re-based.

**Comment by hjoliver** [comment]
Review 1 - good, tests pass.

**Comment by arjclark** [comment]
Looks good to me. Test battery passing in my environment as is manual testing with the example in #1778.

**Pull request merged**